doc/migrating-configs.md

```diff
@@ -2,544 +2,130 @@
-    "config": {
-      "append": [{
-        "source": "https://s3.com/securely-fetched-config.ign"
-      }]
-    },
-    "security": {
-      "tls": {
-        "certificateAuthorities": [
-          {
-            "source": "http://www.example.com/root.pem",
-            "verification": {
-              "hash": "sha512-ab800f66a7544c0a8dbed0c57b38a3c1487c3369e2e9e90704d0c07743557ab2a28c528720566ffc64e3dfd5df1a557a4979b33009f5fd493fea02a7e30041d2"
-            }
-          }
-        ]
-      }
-    }
-  }
-}
-```
-
-### networkd dropins
-
-With the release of systemd v232, networkd dropins are now supported as a means of configuring existing networkd units. The `networkd` section has gained a `dropins` field to reflect this.
-
-```json ignition
-{
-  "ignition": {
-    "version": "2.2.0-experimental"
-  },
-  "networkd": {
-    "units": [{
-      "name": "zz-default.network",
-      "dropins": [{
-        "name": "disable-dhcp.conf",
-        "contents": "data:,%5BNetwork%5D%0ADHCP%3Dno"
-      }]
+      "path": "/var",
+      "device": "/dev/disk/by-label/VAR",
+      "format": "xfs",
+      "wipeFilesystem": true,
+      "label": "var"
+    }],
+    "files": [{
+      "path": "/var/example-file",
+      "mode": 420,
+      "contents": { "source": "" }
     }]
   }
 }
 ```
 
-## From Version 2.0.0 to 2.1.0
+### Files now default to overwrite=false
 
-There are not any breaking changes between versions 2.0.0 and versions 2.1.0 of the configuration specification. Any valid 2.0.0 configuration can be updated to a 2.1.0 configuration by simply changing the version string in the config.
+Files do not overwrite existing files by default. If no source is specified, Ignition will simply ensure a file exists at that path, creating an empty file if necessary.
 
-The 2.1.0 version of the configuration is greatly improved over version 2.0.0, with many new fields and behaviors added to the specification.
+### Directories now create leading directories with mode 0755 and uid/gid 0
 
-The following is a list of notable new features, deprecations, and changes.
+When a file, directory, or link is specified but the parent directory does not exist, Ignition will create the directory(ies) leading up to it. Previously, when a directory was specified, any leading directories were created with the same mode/uid/gid as the specified directory. Now they are created with mode 0755, uid 0, and gid 0. This ensures the behavior is consistent with files and links.
 
-### HTTP timeouts
+### Duplicate entries are no longer allowed
 
-The values used to control the backoff when retrying in HTTP requests can now be set in the config. For details on how the backoff logic works, please see the section in the [operator's notes][operator-notes].
+In order to ensure the configuration file is declarative, duplicates are no longer allowed. This means a config cannot contain two file entries with the same `path`, or specify a path is both a link and a file.
```

Comment:
` In order to ensure the configuration file is declarative` might create more confusion than it solves.

`path as both`